### DISCUSSION TITLE
Offline Mode?? Plus Privacy Concerns!!

### DISCUSSION THREAD

**Discussion by nine1one** [wladradchenko/wunjo.wladradchenko.ru#52] [General]
- App startup time increased if connected to a network.
- When network is disconnect the application srarts in less than 3 seconds.
- Suspicious Network activity: Keeps on doing Check model even in offline mode.
- Not using any audio but the app sends audio for translation in offline mode. I found this after I monitored the traffic and manually blocked a couple of activities by the app.
- Only when there is no network detected by wwunj, the frue speed of the app gets revealed.
- Few other actions and their description or justifications do not align.

Thank You
--

**Comment by wladradchenko** [comment]
Hello, thank you for bringing up these concerns. Let me address each point in detail:

a) Network Requests and Offline Mode: In the current version, the application may make repeated requests to the server when the network is unstable to ensure file downloads are successful. I've introduced an offline mode which can be activated by clicking the planet icon in the app. You can review this [behavior](https://github.com/wladradchenko/wunjo.wladradchenko.ru/blob/main/portable/src/backend/download.py) in the source code here for more details.

b) Suspicious Network Activity: The frequent model [checks](https://github.com/wladradchenko/wunjo.wladradchenko.ru/tree/main/models) are essential for ensuring that models are updated and downloaded correctly. However, I've optimized this logic in the second version of the app. I appreciate your feedback and will continue to enhance the performance and user experience in future updates.

c) Audio Transmission in Offline Mode: Regarding the audio transmission issue, this is unexpected behavior and not intended for offline mode. It's possible there might be a browser-related setting causing this. Please ensure microphone access is disabled when not required.

d) Description Alignment: I've addressed inconsistencies in action descriptions in the second version of the app.

**Reply by nine1one** [reply]
Hi, thank you for your quick response.

I went through the recent commits and I saw what was added, I agree with the additional filtering but

1. The initial checks for models, .py, and .jsons have increased the startup time. (Still exists)

2. I have used various network monitors to confirm network activity post offline mode. They still exist and all the URLs, and network activity suggest they are implementing the filter.

3. The changes took place under the hood without any intimation. That's what I found concerning.  The last release was on Dec 2023 and I did not update the app 1.6.2.

4. The difference in the app performance with or without a network is huge. When the app is used with all network connections disabled, it just works on a boost mode. 

- My intent mainly was to inform you about the delayed start and lags I noticed.

- Upon investigation, found the difference in the app's overall performance with or without a network connection.
----------

Ideas & Suggestions:

1. DARK Mode toggle in the web GUI.

2. Ability to host the service/instance over the network. Then It can be controlled via a browser on any device.

**Reply by wladradchenko** [reply]
I would suggest returning to the discussion after the launch of the second version, since I will not be finalizing the first version, and in the second, some of the problems have been fixed, and some of the problems may not be so critical.  Thanks for your recommendations
